src/Crypto/WebAuthn/Operations/Attestation/AndroidSafetyNet.hs

```diff
@@ -0,0 +1,201 @@
+    unless (commonName == Just hostName) . throwError $ JOSE.JWSInvalidSignature
+
+    -- Create a JWK from the leaf certificate, which is used to sign the payload
+    NE.toList <$> mapM JOSE.fromX509Certificate chain
+
+instance M.AttestationStatementFormat Format where
+  type AttStmt Format = Statement
+  asfIdentifier _ = "android-safetynet"
+
+  type AttStmtDecodingError Format = DecodingError
+  asfDecode _ xs =
+    case (xs !? "ver", xs !? "response") of
+      (Just (TString ver), Just (TBytes responseRaw)) -> do
+        jws :: JOSE.CompactJWS JOSE.JWSHeader <- first DecodingErrorJOSEError $ runExcept $ JOSE.decodeCompact (LB.fromStrict responseRaw)
+        response <-
+          first DecodingErrorJOSEError $
+            JOSE.verifyJWSWithPayload
+              (first JOSE.JSONDecodeError . Aeson.eitherDecode)
+              (JOSE.defaultJWTValidationSettings (const True))
+              androidHostName
+              jws
+        x5c <- extractX5C jws
+        pure $ Statement {..}
+      _ -> Left (DecodingErrorUnexpectedCBORStructure xs)
+    where
+      extractX5C :: JOSE.CompactJWS JOSE.JWSHeader -> Either DecodingError (NE.NonEmpty X509.SignedCertificate)
+      extractX5C jws = do
+        sig <- maybe (Left DecodingErrorJWSMissingSignature) pure $ jws ^? JOSE.signatures
+        JOSE.HeaderParam () x5c <- maybe (Left DecodingErrorJWSMissingX5C) pure $ sig ^. JOSE.header . JOSE.x5c
+        pure x5c
+
+  asfEncode _ Statement {ver, responseRaw} =
+    CBOR.TMap
+      [ (TString "ver", TString ver),
+        (TString "response", TBytes responseRaw)
+      ]
+
+  type AttStmtVerificationError Format = VerificationError
+  asfVerify _ now Statement {..} M.AuthenticatorData {adRawData = M.WithRaw rawData} clientDataHash = do
+    -- 1. Verify that attStmt is valid CBOR conforming to the syntax defined above and perform CBOR decoding on it to
+    -- extract the contained fields.
+    -- NOTE: Done in decoding
+
+    -- 2. Verify that response is a valid SafetyNet response of version ver by following the steps indicated by the
+    -- SafetyNet online documentation. As of this writing, there is only one format of the SafetyNet response and ver
+    -- is reserved for future use.
+    -- NOTE: As stated above, only one version exists, which we assume during decoding
+
+    -- 3. Verify that the nonce attribute in the payload of response is identical to the Base64 encoding of the SHA-256
+    -- hash of the concatenation of authenticatorData and clientDataHash.
+    let signedData = rawData <> BA.convert (M.unClientDataHash clientDataHash)
+    let hashedData = Hash.hashWith Hash.SHA256 signedData
+    let encodedData = Base64.encode $ BA.convert hashedData
+    let nonceBS = BSC.pack $ nonce response
+    unless (nonceBS == encodedData) . Left $ VerificationErrorInvalidSignature
```

Comment:
This uses `BSC.pack` which seems to be a bit weird, as it interprets text, which might have an encoding, as just bytes, and then compares that to other bytes. With `nonce` being a `Text`, I think it's better to go the other way: Encode the `encodedData` as UTF-8 to get a `Text`, then compare that.

```suggestion
    let encodedData = encodeUtf8 . Base64.encode $ BA.convert hashedData
    unless (nonce response == encodedData) . Left $ VerificationErrorInvalidSignature
```